### PR TITLE
Feature/events

### PR DESCRIPTION
**Pull request by tsubik**

### CODE REVIEW

**Comment by kowal** [comment]
There is a difference in order of tabs between show and edit pages:

#### Show:
Zrzut ekranu 2019-08-22 o 11 59 39

#### Edit:
Zrzut ekranu 2019-08-22 o 11 59 46

For me the order could be: `Details - Sides - Events - (Documents)` on both add/edit page.

File: app/models/event.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+class Event < ApplicationRecord
+  belongs_to :eventable, polymorphic: true
+
+  validates_presence_of :title, :date
+  validates :url, url: true
+
+  validates :event_type, presence: true, inclusion: {in: :valid_types}
+
+  def valid_types
+    return Litigation::EVENT_TYPES if eventable.is_a?(Litigation)
```
[the remark this answers is not in the document]
Comment (tsubik) [reply app/models/event.rb]: Validation must be here I guess? but I could add event_types to eventable model sure that's a good idea :+1: but that method must be implemented by all eventables.

File: app/models/litigation.rb [hunk 2]
```diff
@@ -37,10 +43,12 @@ class Litigation < ApplicationRecord
   belongs_to :jurisdiction, class_name: 'Geography'
   has_many :litigation_sides, -> { order(:side_type) }, inverse_of: :litigation
   has_many :documents, as: :documentable, dependent: :destroy
+  has_many :events, as: :eventable, dependent: :destroy
   has_and_belongs_to_many :legislations
 
-  accepts_nested_attributes_for :documents, allow_destroy: true
-  accepts_nested_attributes_for :litigation_sides, allow_destroy: true
+  accepts_nested_attributes_for :documents, allow_destroy: true, reject_if: :all_blank
```
[the remark this answers is not in the document]
Comment (tsubik) [reply app/models/litigation.rb]: :+1:

File: app/helpers/select_helper.rb [hunk 3]
```diff
@@ -1,8 +1,8 @@
 module SelectHelper
-  def array_to_select_collection(array)
+  def array_to_select_collection(array, transform_func = :humanize)
     return unless array.respond_to?(:map)
 
-    array.map { |s| [s.humanize, s] }
+    array.map { |s| [s.send(transform_func), s] }
```
[the remark this answers is not in the document]
Comment (tsubik) [reply app/helpers/select_helper.rb]: ok

**Comment by tsubik** [comment]
Ready again @kowal

**Pull request merged**